Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -482,22 +482,36 @@ namespace AzToolsFramework
             AZStd::unique_ptr<Instance>& sourceInstance, TemplateId targetTemplateId, UndoSystem::URSequencePoint* undoBatch)
         {
             LinkReference nestedInstanceLink = m_prefabSystemComponentInterface->FindLink(sourceInstance->GetLinkId());
-            AZ_Assert(
-                nestedInstanceLink.has_value(),
-                "A valid link was not found for one of the instances provided as input for the CreatePrefab operation.");    
+            if (!nestedInstanceLink)
+            {
+                AZ_Assert(
```

Comment:
This looks like an data error, not a coding error. Why is this not an AZ_Error() with instructions for how to fix the data?

Reply:
Links are a runtime object. If a link is missing, it's a coding error in the Prefab loading code that can't be fixed in data.

Reply:
actually this change probably shouldn't be here. It prevents the crash from happening, but instead there's a bunch of other funky behavior happening otherwise. We want the editor to crash here not to mess up level data later. But it looks like AZ_Assert is getting totally ignored and crash error is unclear.

Reply:
or the function should return a bool, but we need to update all references to handle failure gracefully